src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1009,13 +1009,7 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
 
             var methodResult = result.ValidResult;
             var returnType = methodResult.Member.ReturnType;
-            this.CoerceArguments(methodResult, analyzedArguments.Arguments, diagnostics);
```

Comment:
Any reason for re-arranging these lines?

Reply:
Yes: `CoerceArguments` needs the receiver info.